### ISSUE TITLE
Routing not working properly while using `ng serve` (1.7.0-beta.2)

### ISSUE THREAD

**Issue by MrCroft** [angular/angular-cli#9401]
### Versions
```
Angular CLI: 1.7.0-beta.2 (also happened in beta.1)
Node: 8.9.4
OS: win32 x64
Angular: 5.2.2
```

### Repro steps
* ng new cli-test --routing
* ng g c home
* ng g c about
* edit `app-routing.module.ts` to add the routes for the newly created components:
```
import { HomeComponent } from './home/home.component';
import { AboutComponent } from './about/about.component';

const routes: Routes = [
  { path: 'about', component: AboutComponent },
  { path: '', pathMatch: 'full', component: HomeComponent }
];
```
* edit `app.component.html`:
```
<a [routerLink]="['/']">Home</a>
<a [routerLink]="['/about']">About</a>

<router-outlet></router-outlet>
```
* ng serve
* access localhost:4200
* click on the `About` link. Url becomes `http://localhost:4200/about`
* Refresh the page. Now the HomeComponent is loaded instead of the AboutComponent.

### Observed behavior
After refreshing the page on the `/about` route (or directly accessing the app on `http://localhost:4200/about`), the url turns into `http://localhost:4200/about/` (trailing `/` is added) and the HomeComponent is loaded instead of the AboutComponent. From this point on, any link clicked adds to `/about/` (if we had more than '' and '/about')

**Comment by istiti** [comment]
i have problems with routes too in 1.7.0beta2 site only works on home and navigate into 2/3 deep lazyloading then error like `Loading chunk comptabilite.module failed.` and same problem when refresh on `localhost:4200` it works but refresh on `/+1lazyload/detail` failed all scripts *.bundle.js ERR_ABORTED

**Comment by sbologescu** [comment]
I am having problems in 1.7.0-beta.1
I noticed that routes other than home sometimes get appended to home.
For example, a route that is supposed to go from localhost/home to localhost/some_route ends up to localhost/home/some_route

Like the other users above, refresh on routes with parameters fail. Very annoying since I have to go back to home and navigate to the page I work on once I change code. :(

**Comment by arsyed04** [comment]
I second that. Although 1.7.0-beta.1 and beta.2 have fixed tons of issue regarding ng build times, and errors, ng serve has issue with routes.

**Comment by mruanova** [comment]
That's because when the client makes a request for /merchants, the server needs to return the index.html for the angular routing to kick in. The way to do that is with IIS URL rewriting by adding to your web.config :

<system.webServer>
  <rewrite>
    <rules>
      <rule name="Angular Routes" stopProcessing="true">
        <match url=".*" />
      <conditions logicalGrouping="MatchAll">
        <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
        <add input="{REQUEST_FILENAME}" matchType="IsDirectory" negate="true" />
      </conditions>
      <action type="Rewrite" url="/" />
      </rule>
    </rules>
  </rewrite>
</system.webServer>
This tells iis to rewrite all request that are not valid files or directory to the / url.
Source: https://stackoverflow.com/questions/51159715/angular-6-routing-works-with-ng-serve-not-visual-studio

**Comment by MrCroft** [comment]
@mruanova that's not what this issue was about. It wasn't about the initial request/page load, but about client side navigation specifically. Anyway, it's long solved and closed, right after 1.7.0-beta.2 I think.
